recipes/folly/all/conanfile.py

```diff
@@ -92,23 +84,41 @@ def requirements(self):
         if self.settings.os == "Linux":
             self.requires("libiberty/9.1.0")
             self.requires("libunwind/1.5.0")
-        if Version(self.version) >= "2020.08.10.00":
+        if "2020.08.10.00" <= Version(self.version) < "2022.01.31.00":
             self.requires("fmt/7.1.3")
+        if Version(self.version) >= "2022.01.31.00":
+            self.requires("fmt/8.0.1")  # Folly bump fmt to 8.0.1 in v2022.01.31.00
 
     @property
     def _required_boost_components(self):
         return ["context", "filesystem", "program_options", "regex", "system", "thread"]
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, self._minimum_cpp_standard)
-        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
-        if not min_version:
-            self.output.warn("{} recipe lacks information about the {} compiler support.".format(self.name, self.settings.compiler))
-        else:
-            if Version(self.settings.compiler.version) < min_version:
-                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
-                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
+        compiler_cppstd = self.settings.compiler.get_safe("cppstd")
+        if compiler_cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+        if not compiler_cppstd:
+            def less_than(lhs, rhs):
+                def extract_cpp_version(_cppstd):
+                    return str(_cppstd).replace("gnu", "")
+
+                def add_millennium(_cppstd):
+                    return "19%s" % _cppstd if _cppstd == "98" else "20%s" % _cppstd
+
+                lhs = add_millennium(extract_cpp_version(lhs))
+                rhs = add_millennium(extract_cpp_version(rhs))
+                return lhs < rhs
+            cppstd = default_cppstd(self)
+            if less_than(cppstd, self._min_cppstd):
```

Comment:
Ideally this check should be performed by `check_min_cppstd`. Currently it's not possible because if compiler.cppstd is not set it directly invalidates the configuration but the compiler default cppstd might still be good to go,

Reply:
The common practice is layer out in the templates it's cppstd + compiler version :)

Reply:
Yes i know that. My point is that folly is a "bit" special and requires the user to provide CXX_STD/MSVC_LANGUAGE_VERSION flags with the c++ flag. If the user provides `compiler.cppstd` all is fine, whatever the user provides is what folly will receive. Otherwise we must use `conan.tools.build.default_cppstd` to discover the default cppstd. 

If we follow the common practice of validating compiler.cppstd + compiler version there are configurations where the detected default cppstd violates the _min_cppstd:

default cppstd
```
$ gcc-5 -dM -E -x c++  /dev/null | grep -F __cplusplus
#define __cplusplus 199711L                                           <----- folly would fail if compiler.cppstd is not provided
$ gcc-6 -dM -E -x c++  /dev/null | grep -F __cplusplus
#define __cplusplus 201402L
```
while recipe supports gcc 5:
```
    def _compilers_minimum_version(self):
        return {
            "Visual Studio": "15",
            "gcc": "5",
            "clang": "6",
            "apple-clang": "8",
        }
```
Here it would be incorrect to raise the minimum supported compiler to gcc 6, because users will be perfectly fine using gcc 5 + `-s compiler.cppstd=14`.

So these cases must be filter out. And my second point was that `check_min_cppstd` would be a nice place to do this. For instance, could be extended providing an additional parameter `default_cppstd`, and if compiler.cppstd is not provided, but default_cppstd is, then check lower_than(default_cppstd, min_cppstd). But atm, seems only folly needs this kind of check so maybe it's too soon to generalize :-)

Reply:
Ahhh, that's make more sense.

So in Conan 2.0 this setting is always required. And the detection is the default you are referring so essentially that's the behavior

For 1.x and CCI this history is complicated. The recommendation for now is to `get_safe` and then pass the default as the minimum the project requires and plus the usual so the compiler support is guaranteed  


I agree Folly is wierd but that's the best suggestion I have :)